### PR TITLE
🧹 General clean-up

### PR DESCRIPTION
**Pull request by marekkirejczyk**

### CODE REVIEW

File: packages/example/src/index.tsx [hunk 1]
```diff
@@ -1,11 +1,10 @@
 import React from 'react'
 import ReactDOM from 'react-dom'
-import { ChainId, DAppProvider } from '@usedapp/core'
+import { ChainId, DAppProvider, Config } from '@usedapp/core'
 import { App } from './App'
-import type { Config } from '@usedapp/core/dist/src/model/Config'
 
-const config: Config = {
-  readOnlyChain: ChainId.Mainnet,
+const config: Partial<Config> = {
```
Comment (rzadp) [inline packages/example/src/index.tsx:6]: I don't like that the user has to change our type with `Partial` here.
Also it prevents us from adding really required fields (without a default value) in the future.

How about making the fields in `Config` optional, and maybe having e.g. `ConfigInternal` with fields not-optional (after supplementing with `DEFAULT_VALUE`)?

Comment (marekkirejczyk) [reply packages/example/src/index.tsx]: ✅

File: packages/core/src/providers/ReadOnlyProviderActivator.tsx [hunk 2]
```diff
@@ -1,22 +1,21 @@
...
   const { activate, account, chainId: connectedChainId, active, connector } = useEthers()
-
   useEffect(() => {
-    if (!active || (connector instanceof NetworkConnector && connectedChainId !== chainId)) {
-      activate(new NetworkConnector({ defaultChainId: chainId, urls: nodeUrls }))
+    if (!active || (connector instanceof NetworkConnector && connectedChainId !== readOnlyChainId)) {
+      activate(new NetworkConnector({ defaultChainId: readOnlyChainId, urls: readOnlyUrls || [] }))
     }
-  }, [chainId, active, account, connectedChainId, connector])
+  }, [active, account, connectedChainId, connector])
```
[the remark this answers is not in the document]
Comment (marekkirejczyk) [reply packages/core/src/providers/ReadOnlyProviderActivator.tsx]: ✅

[the remark this answers is not in the document]
Comment (marekkirejczyk) [reply packages/core/src/providers/ReadOnlyProviderActivator.tsx:18]: I tired adding `eslint-plugin-react-hooks`, however:
- it resulted in false positives in several places where code is tricky
- generated a lot of noise in repository (had to fix a lot of new linter rules)
- linter started failing randomly (works locally, but not on CI)

Perhaps someone with better knowledge of linter would like to take on a challenge in a new PR.

**Review by rzadp — APPROVED** [review]

**Pull request merged**